net.sourceforge.vrapper.core.tests/src/net/sourceforge/vrapper/core/tests/cases/SwitchBufferCommandTests.java

```diff
@@ -0,0 +1,208 @@
+package net.sourceforge.vrapper.core.tests.cases;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Arrays;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import net.sourceforge.vrapper.core.tests.utils.VimTestCase;
+import net.sourceforge.vrapper.platform.Buffer;
+import net.sourceforge.vrapper.vim.commands.CommandExecutionException;
+import net.sourceforge.vrapper.vim.commands.SwitchBufferCommand;
+import net.sourceforge.vrapper.vim.modes.NormalMode;
+
+/**
+ * Test case for {@link SwitchBufferCommand}.
+ *
+ */
+public class SwitchBufferCommandTests extends VimTestCase {
+
+    @Before
+    public void activateNormalMode() {
+        adaptor.changeModeSafely(NormalMode.NAME);
+    }
+
+    /**
+     * Test that ":buffer 2" switch to buffer whose ID is 2
+     * 
+     * @throws CommandExecutionException
+     */
+    @Test
+    public void testSwitchNumber() throws CommandExecutionException {
+        final int expectedBufferId = 2;
+        
+        final Buffer buffer1 = mock(Buffer.class);
+        when(buffer1.getId()).thenReturn(1);
+        
+        final Buffer buffer2 = mock(Buffer.class);
+        when(buffer2.getId()).thenReturn(2);
+        
+        when(bufferAndTabsService.getBuffers()).thenReturn(Arrays.asList(buffer1, buffer2));
+        
+        final SwitchBufferCommand cmd = new SwitchBufferCommand(String.valueOf(expectedBufferId));
+        cmd.execute(adaptor, expectedBufferId);
+        
+        verify(bufferAndTabsService).switchBuffer(buffer2);
+    }
+
+    /**
+     * Test that ":buffer buffer2" switch to the buffer whose is buffer2
```

Comment:
Typo, seems the word "name" is missing (here and in the next comments):
```suggestion
     * Test that ":buffer buffer2" switches to buffer with name "buffer2"
```